google-cloud-error_reporting/docs/authentication.md

```diff
@@ -0,0 +1,43 @@
+## With `google-cloud-ruby`
+
+With `google-cloud-ruby` it's incredibly easy to get authenticated and start using Google's APIs. You can set your credentials on a global basis as well as on a per-API basis.
+
+### Project and Credential Lookup
+
+Gcloud aims to make authentication as simple as possible, and provides several mechanisms to configure your system without providing **Project ID** and **Service Account Credentials** directly in code.
```

Comment:
@quartzmo Yeah, this authentication.md is yet-another-copy of the same file from all the other google-cloud-ruby libraries. We should probably fix them all together.

@dazuma I try to keep it consistent in this authenication.md, so users don't get confused when they notice a difference in these authentication instructions (one of the feedback from Monitoring customer). But I added a new section for `Instrumentation Configuration` to show additional way to provide auth info.